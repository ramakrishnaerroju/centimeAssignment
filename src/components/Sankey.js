import React from "react";
import { Chart } from "react-google-charts";
import Config from "./../app.json";

const Sankey = ({ data }) => {
  return (
    <>
      <h2>{Config.sankeyHeader}</h2>
      <Chart
        width={600}
        height={"600px"}
        chartType="Sankey"
        loader={<div>Loading Chart</div>}
        options={{
          sankey: {
            link: { color: { fill: "#00FFFF" } },
            node: {
              colors: ["#348781"],
              label: { color: "#871b47" },
            },
          },
        }}
        data={data}
        rootProps={{ "data-testid": "3" }}
      />
    </>
  );
};

export default Sankey;
