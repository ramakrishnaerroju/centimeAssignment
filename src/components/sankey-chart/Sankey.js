import React from "react";
import { Chart } from "react-google-charts";
import { useTranslation } from "react-i18next";
import PropTypes from "prop-types";

const Sankey = ({ data }) => {
  const { t } = useTranslation();
  return (
    <>
      <h2>{t("sankeyHeader")}</h2>
      <Chart
        width={600}
        height={"600px"}
        chartType="Sankey"
        loader={<div>Loading Chart</div>}
        options={{
          sankey: {
            link: { color: { fill: "#00FFFF" } },
            node: {
              colors: ["#348781"],
              label: { color: "#871b47" },
            },
          },
        }}
        data={data}
      />
    </>
  );
};

Sankey.propTypes = {
  data: PropTypes.arrayOf(PropTypes.any).isRequired,
};

export default Sankey;
