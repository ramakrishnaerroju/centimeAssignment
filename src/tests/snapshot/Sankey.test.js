import React from "react";
import renderer from "react-test-renderer";
import Sankey from "../../components/sankey-chart/Sankey";
import mockdata from "../../assets/mock.json";

test("Sankey chart snapshot test", () => {
  const component = renderer.create(<Sankey data={mockdata["data"]}></Sankey>);
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
