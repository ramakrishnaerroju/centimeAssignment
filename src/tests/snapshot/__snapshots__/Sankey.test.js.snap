// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Sankey chart snapshot test 1`] = `
Array [
  <h2>
    sankeyHeader
  </h2>,
  <div>
    Loading Chart
  </div>,
]
`;
